src/main/java/games/strategy/engine/framework/map/download/DownloadFileParser.java

```diff
@@ -26,38 +27,31 @@ private DownloadFileParser() {}
   }
 
   public static List<DownloadFileDescription> parse(final InputStream is) {
-    @SuppressWarnings("unchecked")
-    final List<Map<String, String>> yamlData = new Yaml().loadAs(is, List.class);
+    final JSONArray yamlData = new JSONArray(new Yaml().loadAs(is, List.class));
 
     final List<DownloadFileDescription> rVal = new ArrayList<>();
-    for (final Map<String, String> yaml : yamlData) {
-      final String url = yaml.get(Tags.url.toString());
-      final String description = yaml.get(Tags.description.toString());
-      final String mapName = yaml.get(Tags.mapName.toString());
-
-      Version version = null;
-      final Object versionObject = yaml.get(Tags.version.toString());
-      if (versionObject != null) {
-        version = new Version(String.valueOf(versionObject));
-      }
-
+    for (int i = 0; i < yamlData.length(); i++) {
+      final JSONObject yaml = yamlData.getJSONObject(i);
+      final String url = yaml.getString(Tags.url.toString());
+      final String description = yaml.getString(Tags.description.toString());
+      final String mapName = yaml.getString(Tags.mapName.toString());
+
+      final Object versionObject = yaml.opt(Tags.version.toString());
+      final Version version = versionObject != null ? new Version(versionObject.toString()) : null;
       DownloadFileDescription.DownloadType downloadType = DownloadFileDescription.DownloadType.MAP;
 
-      final String mapTypeString = yaml.get(Tags.mapType.toString());
+      final String mapTypeString = yaml.optString(Tags.mapType.toString(), null);
```

Comment:
An interesting 3 lines of code here beginning with the `null` default. I think this is a good place to step up to a higher level and examine what we have. The reason for this move was to get type safety, right? Forcing things to be an 'optString' and null defaults I don't think gets us that. Looking at this code:

```
final String mapTypeString = yaml.optString(Tags.mapType.toString(), null);
       if (mapTypeString != null) {
          downloadType = DownloadFileDescription.DownloadType.valueOf(mapTypeString);
        }
```
It's really the same as:
```
final String mapTypeString = yaml.get(Tags.mapType.toString());
       if (mapTypeString != null) {
          downloadType = DownloadFileDescription.DownloadType.valueOf(mapTypeString);
        }
```

I suggest then we should try to write something like:

```
DownloadFileDescription.DownloadType downloadType = yaml.optEnum(Tags.mapType, DownloadFileDescription.DownloadType.MAP);
```

To get there, that would imply to me that we wrap our yaml parsing code in a small library and use that here instead to write similar but simpler and more direct code.

This suggestion of course is not necessary. I am making it though as I feel we get some value from the udpate in this branch, but unless we can write a convincingly similar code, that we have not really have had the impact we want. 

(reference, I very much recommend watching this talk on pythonic vs non-pythonic code: https://www.youtube.com/watch?v=OSGv2VnC0go, the transformation I'm suggesting is in part inspired by that talk)

Reply:
:+1: Good call on using `JSONObject#optEnum()`!  I tried the following on this branch, and it works beautifully:

```java
final DownloadFileDescription.DownloadType downloadType = yaml.optEnum(
    DownloadFileDescription.DownloadType.class,
    Tags.mapType.toString(),
    DownloadFileDescription.DownloadType.MAP);
final DownloadFileDescription.MapCategory mapCategory = yaml.optEnum(
    DownloadFileDescription.MapCategory.class,
    Tags.mapCategory.toString(),
    DownloadFileDescription.MapCategory.EXPERIMENTAL);
```

Reply:
I didn't know that was an actual method, heh, good to see it exists :+1: